pyccel/codegen/printing/fcode.py

```diff
@@ -1740,7 +1711,7 @@ def _print_FunctionDef(self, expr):
         decs.update(self._get_external_declarations())
 
         arguments = [a.var for a in expr.arguments]
-        vars_to_print = self.parser.get_variables(self._namespace)
+        vars_to_print = self.namespace.variables.values()
         for v in vars_to_print:
             if (v not in expr.local_vars) and (v not in expr.results) and (v not in arguments):
                 decs[v] = Declare(v.dtype,v)
```

Comment:
I wonder if it would be clearer to modify `FunctionDef.local_vars` from:
```python
    @property
    def local_vars(self):
        """ List of variables defined in the function """
        return self._local_vars
```
to:
```python
    @property
    def local_vars(self):
        """ List of variables defined in the function """
        local_vars = self.scope.variables.values()
        argument_vars = [a.var for a in expr.arguments]
        return [l for l in local_vars if l not in self.results and l not in argument_vars]
```

Reply:
I think so. But that depends on what one needs this property for...

Reply:
My understanding is that it is used to know which declarations to print. It is also used for the complexity but this doesn't work anyway until #840 is merged (and it could easily collect the current definition by doing `python_local_vars = [v for v in func.local_vars if not v.is_temp]`

Reply:
By the way... in the second code snippet `expr == self`, correct?

Reply:
Oops, yes. Corrected

Reply:
I agree with your suggestion, then.

To improve efficiency it might be useful to compute the `set` union between `argument_vars` and `self.results` before the return statement. It is worth checking different solutions there.

Reply:
Fixed [5e898f5](https://github.com/pyccel/pyccel/pull/1086/commits/5e898f5ae24c8ebc0f32face50c2e3eb46af7f6e) - [3fcf8ed](https://github.com/pyccel/pyccel/pull/1086/commits/3fcf8edd0fc74fa5c5513ee2b4e904bf2e3c0ef4)